docs/designers-developers/developers/internationalization.md

```diff
@@ -1,47 +1,159 @@
+- `_x( 'Default', 'block style', 'my-text-domain' )` - Translate a certain string with some additional context.
+
+**Note:** Every string displayed to the user should be wrapped in an i18n function.
+
+After all strings in your code is wrapped, the final step is to tell WordPress your JavaScript contains translations, using the [wp_set_script_translations()](https://developer.wordpress.org/reference/functions/wp_set_script_translations/) function.
+
+```php
+<?php
+	...
+
+	// Added in init action
+	// Uses script handle defined in register
+	wp_set_script_translations( 'myguten-script', 'myguten' );
+```
+
+This is all you need to make your plugin JavaScript code translatable.
+
+When you set script translations for a handle WordPress will automatically figure out if a translations file exists on translate.wordpress.org, and if so ensure that it's loaded into wp.i18n before your script runs.  With translate.wordpress.org, plugin authors also do not need to worry about setting up their own infrastructure for translations and can rely on a global community with dozens of active locales.  Read more about [WordPress Translations](https://make.wordpress.org/meta/handbook/documentation/translations/).
+
+## Provide Your Own Translations
+
+You can create and ship your own translations with your plugin, if you have sufficient knowledge of the language(s) you can ensure the translations are available.
+
+### Create Translation File
+
+The translation files must be in the JED 1.x JSON format.
+
+To create a JED translation file, first you need to extract the strings from the text.
+Using the [wp-cli tool](https://wp-cli.org/), you create a `.pot` file using the following command from within your plugin directory:
+
+```
+wp i18n make-pot ./
+```
+
+This will create the file `myguten.pot` a pot file is a template file to be used for translating to different languages. The pot file is made up of translation sets, for example:
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr ""
+```
+
+The `msgid` is the string to be translated, and `msgstr` is the translated string, which in the pot file the msgstr will be empty.
+
+A po file is then created from this template, which is the same format, but the translations filled in. You should copy the file using the language code you are going to translate, this example will use the Esperanto (eo) language:
+
+```
+cp myguten.pot myguten-eo.po
+```
+
+You then go through the `.po` file and add the translation to all the `msgstr` sets:
+
+```
+#: block.js:6
+msgid "Simple Block"
+msgstr "Simpla bloko"
+```
+
+
+The last step to create the translation file is to convert the `myguten-eo.po` to the JSON format needed. For this, you use the [po2json utility](https://github.com/mikeedwards/po2json) which you can install using npm. It might be easiest to install globally using: `npm install -g po2json`. Once installed, use the following command to convert to JED format:
```

Comment:
Here I would totally recommend using WP-CLI for this instead.

It's used in the previous step already and supports JSON extraction as well (`wp i18n make-json`).

Recommending po2json means yet another tool one has to install and worry about.

Reply:
I definitely would perfer to use the WP-CLI version, but I struggled with i and have not been able to get it to work. I've read the help, tried numerous different ways.

I wondered why the documentation here also referred to po2json: https://make.wordpress.org/core/2018/11/09/new-javascript-i18n-support-in-wordpress/

Reply:
Ticket opened here #14093